src/Tvheadend.cpp

```diff
@@ -2209,20 +2226,26 @@ void CTvheadend::ParseRecordingAddOrUpdate ( htsmsg_t *msg, bool bAdd )
     }
   }
 
-  /* Channel type fallback (in case channel was deleted) */
-  if (!rec.GetChannelType() && m_conn->GetProtocol() >= 25)
+  htsmsg_t *files, *streams;
+  if ((files = htsmsg_get_list(msg, "files")) != NULL)
   {
-    htsmsg_t *files, *streams;
-    if ((files = htsmsg_get_list(msg, "files")) != NULL)
+    htsmsg_field_t *file, *stream;
+    uint32_t u32;
+    int64_t s64;
+    bool needChannelType = !rec.GetChannelType() && m_conn->GetProtocol() >= 25;
```

Comment:
Is there any point in doing this optionally? We used to call `SetChannelType` unconditionally.

Reply:
> We used to call SetChannelType unconditionally.

 Not correct. We called it only if we had no channeltype.

Reply:
was:

<pre>
if (!rec.GetChannelType() && m_conn->GetProtocol() >= 25)
{
  htsmsg_t *files, *streams;
  if ((files = htsmsg_get_list(msg, "files")) != NULL)
...
</pre>

is:

<pre>
htsmsg_t *files, *streams;
if ((files = htsmsg_get_list(msg, "files")) != NULL)
{
...
  bool needChannelType = !rec.GetChannelType() && m_conn->GetProtocol() >= 25;
</pre>

becasue we now need to always parse the "files" map, but still only want to set the channel type according to "files" information if we do not already have a channel type (exactly like before).

Reply:
Sorry, the diff was a bit hard to read

Reply:
np